backend/pkg/api/handle_transforms.go

```diff
@@ -0,0 +1,290 @@
+		if !canList {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("requester has no permissions to list wasm transforms"),
+				Status:   http.StatusForbidden,
+				Message:  "You don't have permissions to list wasm transforms",
+				IsSilent: false,
+			})
+			return
+		}
+
+		if !api.Cfg.Redpanda.AdminAPI.Enabled {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("you must enable the admin api to manage wasm transforms"),
+				Status:   http.StatusBadRequest,
+				Message:  "you must enable the admin api to manage wasm transforms",
+				IsSilent: false,
+			})
+			return
+		}
+
+		transforms, err := api.RedpandaSvc.ListWasmTransforms(r.Context())
+		if err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusInternalServerError,
+				Message:  "Could not list wasm transforms from Kafka cluster",
+				IsSilent: false,
+			})
+			return
+		}
+
+		rest.SendResponse(w, r, api.Logger, http.StatusOK, response{
+			Transforms: transforms,
+		})
+	}
+}
+
+// DeployTransformRequest defines the expected JSON body to deploy a transform.
+type DeployTransformRequest struct {
+	Name         string                         `json:"name"`
+	InputTopic   string                         `json:"input_topic"`
+	OutputTopics []string                       `json:"output_topics"`
+	Environment  []adminapi.EnvironmentVariable `json:"environment"`
+	WasmBinary   []byte                         `json:"wasm_binary"`
+}
+
+func (api *API) handleDeployTransform() http.HandlerFunc {
+	type response struct {
+		Transform *adminapi.TransformMetadata `json:"transform"`
+	}
+	return func(w http.ResponseWriter, r *http.Request) {
+		canList, restErr := api.Hooks.Authorization.CanDeployTransform(r.Context())
+		if restErr != nil {
+			rest.SendRESTError(w, r, api.Logger, restErr)
+			return
+		}
+		if !canList {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("requester has no permissions to list wasm transforms"),
+				Status:   http.StatusForbidden,
+				Message:  "You don't have permissions to list wasm transforms",
+				IsSilent: false,
+			})
+			return
+		}
+
+		if !api.Cfg.Redpanda.AdminAPI.Enabled {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("you must enable the admin api to manage wasm transforms"),
+				Status:   http.StatusBadRequest,
+				Message:  "you must enable the admin api to manage wasm transforms",
+				IsSilent: false,
+			})
+			return
+		}
+
+		var msg DeployTransformRequest
+		if restErr := rest.Decode(w, r, &msg); restErr != nil {
+			rest.SendRESTError(w, r, api.Logger, restErr)
+			return
+		}
+
+		if err := api.RedpandaSvc.DeployWasmTransform(r.Context(), adminapi.TransformMetadata{
+			Name:         msg.Name,
+			InputTopic:   msg.InputTopic,
+			OutputTopics: msg.OutputTopics,
+			Status:       nil,
+			Environment:  msg.Environment,
+		}, msg.WasmBinary); err != nil {
```

Comment:
FWIW one thing we sanity check WRT WasmBinary in Redpanda (and I think RPK) is making sure the magic bytes for a wasm binary are valid to double check someone didn't try uploading a native binary.

Magic bytes are `\0asm` (null byte followed by ascii `a` `s` `m`).

Reply:
looks like there isn't a check in RPK: https://github.com/redpanda-data/redpanda/blob/821bd6d1fd31ef4efc9c699eade449d40d089568/src/go/rpk/pkg/adminapi/api_transform.go#L24

IMO it'd make more sense to check it in RPK since we're doing the actual send with the admin client from rpk anyway.

Reply:
https://github.com/redpanda-data/redpanda/issues/16230 created an issue

Reply:
and a PR here https://github.com/redpanda-data/redpanda/pull/16231

Reply:
This happens in the deploy command not the validation at the moment.


As for not doing it here and only in the rpk, isn't part of the point of these APIs is so people don't have to use RPK?